### PR TITLE
Fixed an issue that was causing the results of the ctx object to be empty. 

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
1. Fixed an issue that was causing the results of the ctx object to be empty when viewing the available `ctx` variables in the Trigger condition `Info` flyout. 
2. Fixed an issue that was causing variables like `ctx.trigger.name` to not populate in the message preview.

### Known issues
1. ~~https://github.com/opensearch-project/alerting-dashboards-plugin/issues/333 describes an issue that's causing the `ctx.monitor.schedule.period.interval` and `ctx.monitor.schedule.period.unit` to be blank when the notification is sent to the webhook. This is still under investigation. The message preview in the UI **does** show those values, suggesting this may be a backend issue.~~ As discussed in issue https://github.com/opensearch-project/alerting-dashboards-plugin/issues/333, this known issue results from the backend, and there is a PR published to address it.

### Issues Resolved
https://github.com/opensearch-project/alerting-dashboards-plugin/issues/152
https://github.com/opensearch-project/alerting-dashboards-plugin/issues/333

### Screenshots
Screen Shot 2022-09-28 at 9 03 33 PM
Screen Shot 2022-09-28 at 9 04 09 PM
Screen Shot 2022-09-28 at 9 06 36 PM
Screen Shot 2022-09-28 at 9 08 06 PM
Screen Shot 2022-09-28 at 9 11 12 PM
Screen Shot 2022-09-28 at 9 12 57 PM
Screen Shot 2022-09-28 at 9 13 27 PM
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
The cypress tests are failing because the plugin version doesn't match the version of OpenSearch-Dashboards used by the tests. A separate PR will be submitted to adjust the version.

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js [hunk 1]
```diff
@@ -92,8 +87,18 @@ class ConfigureTriggers extends React.Component {
...
+        monitorValues: { uri },
+      } = this.props;
+      switch (monitor_type) {
+        case MONITOR_TYPE.BUCKET_LEVEL:
+          this.onQueryMappings();
+          break;
+        case MONITOR_TYPE.CLUSTER_METRICS:
+          if (canExecuteClusterMetricsMonitor(uri)) this.onRunExecute();
+          break;
+      }
```
Comment (amsiglan) [inline public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js:101]: Same as lines 47 to 58 above. Let's extract into a method

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js [hunk 2]
```diff
@@ -131,7 +136,7 @@ class ConfigureTriggers extends React.Component {
       case SEARCH_TYPE.QUERY:
       case SEARCH_TYPE.GRAPH:
         const searchRequest = buildRequest(formikValues);
-        _.set(monitorToExecute, 'inputs[0].search', searchRequest);
+        _.set(monitorToExecute, 'inputs[0]', searchRequest);
```
Comment (amsiglan) [inline public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js:139]: This change has been checked in as part of [this PR](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/382), so just ensure there are no conflicts

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js (hunk 1 above)
Comment (AWSHurneyt) [reply public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js:101]: Good catch. Consolidated this duplicate code.

**Review by amsiglan — APPROVED** [review]

File: public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js (hunk 2 above)
Comment (AWSHurneyt) [reply public/pages/CreateTrigger/containers/ConfigureTriggers/ConfigureTriggers.js]: Rebasing the recent changes to `main` onto this dev branch has removed this duplicate change from this PR.

**Review by praveensameneni — APPROVED** [review]

**Pull request merged**